python_modules/libraries/dagster-snowflake/dagster_snowflake/snowflake_io_manager.py

```diff
@@ -52,10 +54,19 @@ def snowflake_io_manager():
 class SnowflakeDbClient(DbClient):
     @staticmethod
     def delete_table_slice(context: OutputContext, table_slice: TableSlice) -> None:
+        no_schema_config = (
+            {k: v for k, v in context.resource_config.items() if k != "schema"}
+            if context.resource_config
+            else {}
+        )
         with SnowflakeConnection(
-            dict(schema=table_slice.schema, **(context.resource_config or {})), context.log  # type: ignore
+            dict(schema=table_slice.schema, **no_schema_config), context.log  # type: ignore
         ).get_connection() as con:
-            con.execute_string(_get_cleanup_statement(table_slice))
+            try:
+                con.execute_string(_get_cleanup_statement(table_slice))
+            except ProgrammingError:
```

Comment:
Do we want to be specific here about the text we expect to distinguish from other things that could cause an error? Might not matter, as it's hard to imagine a situation beyond "table doesn't exist" where we'd run into an error here but not when trying to write the data.

Reply:
yeah i didn't have any ideas of errors that wouldn't be caught at the write step. The error message is super simple though, so it'd be easy to match on (thinking "does not exist" would do it)

```
 002003 (42S02): SQL compilation error:
Object 'SANDBOX.JAMIE.STARTER' does not exist or not authorized.
```